### PR TITLE
Developer Portal > Deploy > SAP Cloud Platform

### PR DESCRIPTION
**Pull request by MarkvanMents**
Rewrite of documentation with new screenshots and additional information
Update to Deployment > SAP Cloud Platform to encourage users to use the Developer Portal

### CODE REVIEW

**Comment by MarkvanMents** [comment]
Needs review by DES before merging to development

File: content/deployment/sap-cloud-platform/deploy-a-mendix-app-to-sap-cloud-platform.md [hunk 1]
```diff
@@ -24,7 +26,7 @@ In addition to these required services, the following services are optional:
 * Connectivity Service
 * ObjectStore (S3)
 
-When using the [Mendix Developer Portal](/developerportal/deploy/sap-cloud-platform), these services will be automatically instantiated and bound to the application.
+It is recommended that you use the instructions for deploying through the [Mendix Developer Portal](/developerportal/deploy/sap-cloud-platform). If you do this, these services will be automatically instantiated and bound to the application.
 
 ### 2.1 Trial Account
```
Comment (MehranIr) [inline content/deployment/sap-cloud-platform/deploy-a-mendix-app-to-sap-cloud-platform.md]: This feels like you could only use the xsmall db where as technically they can use whatever postgres db that they can get for their mx app

File: content/deployment/sap-cloud-platform/deploy-a-mendix-app-to-sap-cloud-platform.md [hunk 2]
```diff
@@ -118,14 +120,16 @@ To bind your services to SAP Cloud Platform Cloud Foundry, follow these steps:
 
 8. Enter an **Instance Name** and click **Finish**.
 
-Your application is now set up with a minimum configuration to run your application. Setting up other services follows a similar process. For information about configuring these services, consult the links in [4 Related Content](#RC).
+Your application is now set up with the minimum configuration needed to run your application. Follow a similar process to set up other services. For information about configuring these services, consult the links in section [4 Related Content](#RC).
```
Comment (MehranIr) [inline content/deployment/sap-cloud-platform/deploy-a-mendix-app-to-sap-cloud-platform.md]: don't we also need to do some stuff like setting admin pass?

File: content/deployment/sap-cloud-platform/deploy-a-mendix-app-to-sap-cloud-platform.md (hunk 1 above)
Comment (MarkvanMents) [reply content/deployment/sap-cloud-platform/deploy-a-mendix-app-to-sap-cloud-platform.md]: This is the old document which we are thinking of deleting, so it is not worth making it much better.

File: content/deployment/sap-cloud-platform/deploy-a-mendix-app-to-sap-cloud-platform.md (hunk 2 above)
Comment (MarkvanMents) [reply content/deployment/sap-cloud-platform/deploy-a-mendix-app-to-sap-cloud-platform.md]: This is the old Cloud Foundry document - I'm much more interested in comments on the other document.

File: content/developerportal/deploy/sap-cloud-platform.md [hunk 3]
```diff
@@ -7,117 +7,325 @@ tags: ["SAP", "SAP Cloud Portal", "Deployment", "Environment"]
...
 
-## 3 Create Package from Team Server
+Click **Getting Started** and then continue with section [2.3 Set Up Region](#SetUpRegion).
 
-At any time, it is possible to create a new deployment package from the project.
+### 2.2 New SAP App<a name='NewSAPApp'></a>
 
-1. Click **Create package from Team Server** and start the wizard.
-2.  Select the branch you want to use from the Team Server.
+In this scenario, you choose a Mendix starter app from the SAP Apps tab.
```
Comment (Nidhi251289) [inline content/developerportal/deploy/sap-cloud-platform.md]: This is bit confuisng when you say choose Mendix Starter app from SAP app tabs.. u can may be just mention a  SAP template

File: content/developerportal/deploy/sap-cloud-platform.md [hunk 4]
```diff
@@ -7,117 +7,325 @@ tags: ["SAP", "SAP Cloud Portal", "Deployment", "Environment"]
...
+
+![](attachments/sap-cloud-platform/01-sap-login-no-credentials.png)
+
+You will now be asked to provide the final details for the SAP Cloud Platform development environment.
+
+![](attachments/sap-cloud-platform/create-development.png)
+
+You will be able to choose a Domain, Organization, and Space which is configured for you in this region.
+
+If you do not choose a Custom database, you will still be able to choose from a range of different Postgres databases. Please ensure that the database you choose is supported by your quota plan for this region and organization.
```
Comment (Nidhi251289) [inline content/developerportal/deploy/sap-cloud-platform.md]: In custome database, we have option to provide the service and plan name of the data base. The other option is choosing the services from drop down.

Probably, you can mention that the user can either choose the service from drop down or they can provide the service and plan name for the database

File: content/developerportal/deploy/sap-cloud-platform.md [hunk 5]
```diff
@@ -7,117 +7,325 @@ tags: ["SAP", "SAP Cloud Portal", "Deployment", "Environment"]
...
+
+If you choose a Custom database, you will be asked for the Name and the Plan.
```
Comment (Nidhi251289) [inline content/developerportal/deploy/sap-cloud-platform.md]: If you select Yes for Custom database

File: content/developerportal/deploy/sap-cloud-platform.md [hunk 6]
```diff
@@ -7,117 +7,325 @@ tags: ["SAP", "SAP Cloud Portal", "Deployment", "Environment"]
...
+
+You can create several environments for your app. For example, you may have created a development environment, but you may want environments for test, acceptance, production, and so forth. Additionally, when you switch from another cloud you need to create at least one environment for your Mendix application.
 
-## 4 Create a new Environment
+This is done from the **Environments** page of the Developer Portal:
 
-You need to create at least one environment for each Mendix application. This is done for you when creating an application from an SAP Starter App. When you switch between clouds this is something you need to do yourself.
+![](attachments/sap-cloud-platform/environments-page.png)
 
 1. Click **Add Environment** to start the wizard.
```
Comment (Nidhi251289) [inline content/developerportal/deploy/sap-cloud-platform.md]: You can mention that 
If you want to provide more details in your environment or customize it (lets say give a different name to the environment) in that user can select Add environment.

File: content/developerportal/deploy/sap-cloud-platform.md [hunk 7]
```diff
@@ -7,117 +7,325 @@ tags: ["SAP", "SAP Cloud Portal", "Deployment", "Environment"]
...
-A second environment is created. It is now possible to transport applications between environments.
+An environment is created. With more than one environment it is possible to transport your application between environments (see section [6 Transport App Between Environments](#TransportApp)).
 
 ![](attachments/sap-cloud-platform/10-sap-env-tap.png)
 
-### 4.1 Environment Details
+## 4 Create Package from Team Server
 
-The environment details page contains two tabs.
+At any time, it is possible to create a new deployment package from a committed version of the project. If you are working with Desktop Modeler you will first have to commit the project manually. The Web Modeler will commit versions automatically.
```
Comment (Nidhi251289) [inline content/developerportal/deploy/sap-cloud-platform.md]: If you want to use deployment package from DM, then youhave to first commit it and the latest reviosion you can use in creating deployment package

File: content/developerportal/deploy/sap-cloud-platform.md [hunk 8]
```diff
@@ -7,117 +7,325 @@ tags: ["SAP", "SAP Cloud Portal", "Deployment", "Environment"]
...
+1. You will see confirmation of the package which has been transported.
+
+    ![](attachments/sap-cloud-platform/transport-deploy.png)
+
+2. Change any constants in the Constants tab: select the constant you want to edit and then click **Edit**.
+
+    ![](attachments/sap-cloud-platform/transport-constants.png)
 
-Settings you can change:
+3. Toggle any scheduled events in the Scheduled Events tab: select the scheduled event you want to enable or disable and click **Toggle**.
```
Comment (Nidhi251289) [inline content/developerportal/deploy/sap-cloud-platform.md]: Also you can mention that the changes will be updated once the application is restarted in both constants and Scheduled events

File: content/developerportal/deploy/sap-cloud-platform.md (hunk 3 above)
Comment (MarkvanMents) [reply content/developerportal/deploy/sap-cloud-platform.md]: Covered by the picture, I think

File: content/developerportal/deploy/sap-cloud-platform.md (hunk 6 above)
Comment (MarkvanMents) [reply content/developerportal/deploy/sap-cloud-platform.md]: I thought it said that! "...you may want environments for test, acceptance, production, and so forth."

File: content/developerportal/deploy/sap-cloud-platform.md (hunk 7 above)
Comment (MarkvanMents) [reply content/developerportal/deploy/sap-cloud-platform.md]: Reference to Web Modeler removed

File: content/developerportal/deploy/sap-cloud-platform.md (hunk 8 above)
Comment (MarkvanMents) [reply content/developerportal/deploy/sap-cloud-platform.md]: Not here - you can only deploy if the application is stopped already.

File: content/developerportal/deploy/sap-cloud-platform.md [hunk 9]
```diff
@@ -7,117 +7,325 @@ tags: ["SAP", "SAP Cloud Portal", "Deployment", "Environment"]
...
+
+#### 7.1.3 View Recent Log
+
+Click **View Recent Log** to see recent events written to the log.
+
+#### 7.1.4 Delete Environment
+
+**Delete Environment** enables you to delete the environment, along with all its resources: including the app.
+
+You will be asked to confirm that this environment should be removed. You will also be asked whether the resources associated with the environment should also be removed.
```
[the remark this answers is not in the document]
Comment (MarkvanMents) [reply content/developerportal/deploy/sap-cloud-platform.md]: It does need two confirmations - one is the tick box for resources, the other is the confirmation button.

File: content/developerportal/deploy/sap-cloud-platform.md [hunk 10]
```diff
@@ -7,117 +7,325 @@ tags: ["SAP", "SAP Cloud Portal", "Deployment", "Environment"]
...
+
+Click **Change Admin Password** to change the password for the administrator account (by default, MxAdmin) in your Mendix app.
+
+#### 7.1.3 View Recent Log
+
+Click **View Recent Log** to see recent events written to the log.
+
+#### 7.1.4 Delete Environment
+
+**Delete Environment** enables you to delete the environment, along with all its resources: including the app.
```
[the remark this answers is not in the document]
Comment (MarkvanMents) [reply content/developerportal/deploy/sap-cloud-platform.md]: see below... - have amended the text slightly

**Pull request merged**